File: 백준/Gold/31963. 두 배/두 배.py
import sys
import math

def solution(N, numbers):
    answer = 0
    for i in range(1, len(numbers)):
        prev = numbers[i - 1]
        next = numbers[i]
        if prev > next:
            m = math.ceil(math.log2(prev / next))  # math.log2 사용
            numbers[i] = numbers[i] * 2 ** m  # ** 연산자 사용
            answer += m  # += 연산자 사용
    return answer

# 입력 처리
if __name__ == "__main__":
    N = int(input())
    numbers = list(map(int, input().split()))  # 입력 받아 정수 리스트로 변환
    print(solution(N, numbers))  # 결과 출력

File: 백준/Gold/31963. 두 배/두 배.js
const fs = require('fs');
const filePath = process.platform === "linux" ? "dev/stdin" : "test.txt";
const input = fs.readFileSync(filePath).toString().trim().split('\n');
const N = Number(input[0])
const numbers = input[1].trim().split(' ').map((a)=>Number(a))

/**
 * 
 * @param {Number} N 
 * @param {Array} numbers 
 * @returns 
 */
function solution(N, numbers) {
    let answer = 0
    for (let i = 1; i<numbers.length; i++){
        const prev = numbers[i-1]
        const next = numbers[i]
        if (prev > next){
            const m = Math.ceil(Math.log2(prev/next))
            numbers[i] = numbers[i] * 2 ** m
            answer = answer + m
        }
    }
    return answer;
}

console.log(solution(N, numbers));